### PR TITLE
Added [prefix] argument to DefinePlugin constructor

### PR DESCRIPTION
**Pull request by SerhiiBilyk**
Motivation : fix issue 2129
Link: https://github.com/webpack/webpack/issues/2129

**What kind of change does this PR introduce?**
Bugfix

**Did you add tests for your changes?**
Not yet

**Does this PR introduce a breaking change?**
I think yes

**What needs to be documented once your changes are merged?**
I have created this PL only for question, because I'm stack with purpose of prefix argument.
I have added [prefix] argument to DefinePlugin constructor, but I don't know what expected behavior of this argument. Maybe somebody can provide an example or can explain me. Thank you.

P.S. This is my first work with open source projects, so don't judge me if I did something wrong

### CODE REVIEW

**Comment by SerhiiBilyk** [comment]
@evilebottnawi  Is there other tests of DefinePlugin except Integration.test.js ?

File: lib/DefinePlugin.js [hunk 1]
```diff
@@ -74,9 +74,11 @@ class DefinePlugin {
 	/**
 	 * Create a new define plugin
 	 * @param {Record<string, CodeValue>} definitions A map of global object definitions
+	 * @param {string} prefix prefix
```
Comment (ooflorent) [inline lib/DefinePlugin.js]: Should be `{string=}` since the parameter is optional

**Comment by SerhiiBilyk** [comment]
@ooflorent on my local all tests are passed, but travis-ci has other opinion :). What I'm doing wrong?

**Comment by sokra** [comment]
Why is the PR needed? What's the use case of a `prefix` argument?

**Comment by ooflorent** [comment]
I guess the use case is to _simplify_ how the definition is done:

```js
new DefinePlugin({
  FOO: 1,
  BAR: 2
}, "ENV_")
```

I must admit this is not really needed.

**Comment by SerhiiBilyk** [comment]
@ooflorent , @ sokra ok, then I'm closing this PL

File: test/statsCases/define-plugin/index.js [hunk 2]
```diff
@@ -1 +1,2 @@
 module.exports = VALUE;
+module.exports=prefixVALUE;
```
Comment (montogeek) [inline test/statsCases/define-plugin/index.js]: Please follow code format: Add space before and after the equal.

**Pull request closed without merging**